tensorflow_lite_support/c/task/vision/image_classifier.cc

```diff
@@ -0,0 +1,225 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/vision/image_classifier.h"
+
+#include <memory>
+
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "tensorflow_lite_support/c/task/vision/utils/frame_buffer_cpp_c_utils.h"
+#include "tensorflow_lite_support/cc/task/vision/image_classifier.h"
+#include "tensorflow_lite_support/cc/task/vision/proto/classifications_proto_inc.h"
+#include "tensorflow_lite_support/cc/task/vision/proto/image_classifier_options_proto_inc.h"
+
+namespace {
+using ::tflite::support::StatusOr;
+using ClassificationResultCpp = ::tflite::task::vision::ClassificationResult;
+using ClassificationsCpp = ::tflite::task::vision::Classifications;
+using ClassCpp = ::tflite::task::vision::Class;
+using BoundingBoxCpp = ::tflite::task::vision::BoundingBox;
+using ImageClassifierCpp = ::tflite::task::vision::ImageClassifier;
+using ImageClassifierOptionsCpp =
+    ::tflite::task::vision::ImageClassifierOptions;
+using FrameBufferCpp = ::tflite::task::vision::FrameBuffer;
+}  // namespace
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+struct TfLiteImageClassifier {
+  std::unique_ptr<ImageClassifierCpp> impl;
+};
+
+std::unique_ptr<ImageClassifierOptionsCpp>
+CreateImageClassifierCppOptionsFromCOptions(
+    const TfLiteImageClassifierOptions* c_options) {
+  std::unique_ptr<ImageClassifierOptionsCpp> cpp_options(
+      new ImageClassifierOptionsCpp);
+
+  // More file sources can be added in else ifs
+  if (c_options->base_options.model_file.file_path)
+    cpp_options->mutable_base_options()->mutable_model_file()->set_file_name(
+        c_options->base_options.model_file.file_path);
+  else
+    return nullptr;
+
+  if (c_options->base_options.compute_settings
+                               .tflite_settings.cpu_settings.num_threads > 0)
+              cpp_options->mutable_base_options()
+          ->mutable_compute_settings()
+          ->mutable_tflite_settings()
+          ->mutable_cpu_settings()
+          ->set_num_threads(c_options->base_options.compute_settings
+                                .tflite_settings.cpu_settings.num_threads);
+  else {
+    cpp_options->mutable_base_options()
+        ->mutable_compute_settings()
+        ->mutable_tflite_settings()
+        ->mutable_cpu_settings()
+        ->set_num_threads(-1);
+  }
+
+  if (c_options->classification_options.class_name_blacklist.length > 0 &&
+      c_options->classification_options.class_name_whitelist.length > 0)
+    return nullptr;
+
+  if (c_options->classification_options.class_name_blacklist.length > 0 &&
+      c_options->classification_options.class_name_whitelist.length)
+    return nullptr;
```

Comment:
These checks can be removed, because they will be checked and verified in the C++ layer.

Reply:
Removed for blacklist and whitelist. I tested the CPP implementation with various values of `num_threads`. Please see my reason for not removing the check for ` num_threads` [here]( https://github.com/tensorflow/tflite-support/pull/658#discussion_r704151228). It's a similar scenario.

Reply:
I agree we should implementing a function to return the default settings. Please use the same [default value](https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/cc/task/processor/proto/classification_options.proto) as defined in the C++ layer.

Reply:
And also a function to delete/free the setting struct?

Reply:
Yeah if framework is creating the default struct, a free is needed. Do you want me to update this in this PR. Or once this is merged as most tests would also change. Won't be a very big change though.

Reply:
@lu-wang-g By default variable sized members like class name black list, white list, and strings like display_name_local will be initialised to null pointers when we provide method to return a default initialised struct. In this case, client can allocate them using malloc or just assign address of a variable on stack. If we return a pointer to the options, we will have to provide a free method and we wouldn't know how the pointer members like strings, string arrays were allocated and hence can't simply call a delete on these members if they are passed in to the C library for deallocation. Wouldn't it be better to just return an options struct by value when client calls TfLiteImageClassifierOptionsCreate initialized with default values and avoid providing a free method. We can leave the alloc and free of pointer members to the client like how it is currently and still provide them a default initialised options which will avoid garbage values and redundant checks in C code.

Reply:
Yeah, I think that makes sense. Please create TfLiteImageClassifierOptionsCreate and let it return an options struct object. Thanks for the discussion!